src/containers/ThreadsDropdown.js

```diff
@@ -25,30 +28,75 @@ const getThreads = state =>
     sortBy(t => -1 * new Date(t.updated_at))
   )(state.postsByQuery.threads)
 
-export const ThreadsDropdown = connect(
+@connect(
   (state, props) => ({
     threads: getThreads(state),
     pending: state.pending[FETCH_POSTS]
   })
-)(props => {
-  const { threads, dispatch, pending, newCount } = props
-  return <Dropdown alignRight rivalrous='nav' className='thread-list'
-    onFirstOpen={() => dispatch(fetchPosts({ cacheId: 'threads', subject: 'threads' }))}
-    toggleChildren={<span>
-      <Icon name='Message-Smile'/>
-      {newCount > 0 && <div className='badge'>{newCount}</div>}
-    </span>}>
-    {!pending && <li className='top'>
-      <div className='newMessage' onClick={() => dispatch(showDirectMessage())}>
-        <Icon name='Compose'/><span className='button-text'>New Message</span>
-      </div>
-    </li>}
-    {pending && <li className='loading'>Loading...</li>}
-    {threads.map(thread => <li key={thread.id}>
-      <Thread thread={thread}/>
-    </li>)}
-  </Dropdown>
-})
+)
+export class ThreadsDropdown extends React.Component {
+
+  static propTypes = {
+    threads: array,
+    pending: bool,
+    newCount: number 
+  }
+
+  static contextTypes = {
+    dispatch: func,
+    currentUser: object
+  }
+
+  componentDidMount () {
+    const { dispatch } = this.context
+    this.socket = getSocket()
+    this.socket.post(socketUrl('/noo/threads/subscribe'))
+    this.socket.on('newThread', this.newThread.bind(this))
+    this.socket.on('messageAdded', this.messageAdded.bind(this))
+  }
+
+  componentWillUnmount () {
+    if (this.socket) {
+      this.socket.post(socketUrl('/noo/threads/unsubscribe'))
+      this.socket.off('newThread')
+      this.socket.off('messageAdded')
+    }
+  }
+
+  newThread (thread) {
+    const { dispatch, currentUser } = this.context
+    dispatch(appendThread(thread))
+    dispatch(incrementUnreadThreads()) 
+    this.forceUpdate()
```

Comment:
this shouldn't be necessary -- if you're finding that it is, it probably means one of the reducers is modifying the existing state object rather than replacing it.